gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/services/ProjectAgentService.java

```diff
@@ -1,44 +1,77 @@
+import dev.ctrlspace.gendox.gendoxcoreapi.model.User;
+import dev.ctrlspace.gendox.gendoxcoreapi.model.authentication.JwtDTO;
 import dev.ctrlspace.gendox.gendoxcoreapi.repositories.ProjectAgentRepository;
 import dev.ctrlspace.gendox.gendoxcoreapi.repositories.ProjectRepository;
+import dev.ctrlspace.gendox.gendoxcoreapi.repositories.TemplateRepository;
+import dev.ctrlspace.gendox.gendoxcoreapi.repositories.UserRepository;
+import dev.ctrlspace.gendox.gendoxcoreapi.utils.JWTUtils;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.http.HttpStatus;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.jwt.Jwt;
 import org.springframework.stereotype.Service;
 
 import java.time.Instant;
+import java.util.Optional;
 import java.util.UUID;
 
 @Service
 public class ProjectAgentService {
 
+
+    @Value("${gendox.agents.splitter-type}")
+    private String splitterTypeName;
+
     private ProjectAgentRepository projectAgentRepository;
+    private UserRepository userRepository;
+    private JWTUtils jwtUtils;
+    private TypeService typeService;
+    private TemplateRepository templateRepository;
 
     @Autowired
-    public ProjectAgentService(ProjectAgentRepository projectAgentRepository){
+    public ProjectAgentService(ProjectAgentRepository projectAgentRepository,
+                               UserRepository userRepository,
+                               JWTUtils jwtUtils,
+                               TypeService typeService,
+                               TemplateRepository templateRepository) {
         this.projectAgentRepository = projectAgentRepository;
+        this.userRepository = userRepository;
+        this.jwtUtils = jwtUtils;
+        this.typeService = typeService;
+        this.templateRepository = templateRepository;
     }
 
-    public ProjectAgent createProjectAgent(ProjectAgent projectAgent) throws Exception{
+    public ProjectAgent createProjectAgent(ProjectAgent projectAgent) throws Exception {
         Instant now = Instant.now();
 
-        if(projectAgent.getId() != null){
+        if (projectAgent.getId() != null) {
             throw new GendoxException("NEW_PROJECT_AGENT_ID_IS_NOT_NULL", "Project - Agent id must be null", HttpStatus.BAD_REQUEST);
 
         }
 
         projectAgent.setCreatedAt(now);
         projectAgent.setUpdatedAt(now);
+        projectAgent.setCreatedBy(getUserId());
+        projectAgent.setUpdatedBy(getUserId());
+        projectAgent.setChatTemplateId(templateRepository.findIdByIsDefaultTrueAndTemplateTypeName("CHAT_TEMPLATE"));
+        projectAgent.setSectionTemplateId(templateRepository.findIdByIsDefaultTrueAndTemplateTypeName("SECTION_TEMPLATE"));
+        projectAgent.setDocumentSplitterType(typeService.getDocumentSplitterTypeByName(splitterTypeName));
```

Comment:
Add default values only if the request has nulls. If it doesn't have null fields, add the requested types